examples/test_messaging/test_messaging.c

```diff
@@ -0,0 +1,364 @@
+                                printf("PASSED TEST 2\n");
+                                printf("---------------------------\n");
+                                msg_params->tests_passed++;
+                                msg_params->test_phase++;
+                                msg_params->test_msg_count = 0;
+                        }
+                }
+                rte_free(msg_data);
+                if (0 == msg_params->test_msg_count) {
+                        printf("TEST 3: Message Ring Overflow...\n");
+                        printf("---------------------------\n");
+                        for (int i = 0; i < (int)(rte_ring_get_size(msg_params->msg_q) + 2); i++) {
+                                int* msg_int = (int*)rte_malloc(NULL, sizeof(int), 0);
+                                *msg_int = i;
+                                onvm_nflib_send_msg_to_nf(msg_params->address, msg_int);
+                        }
+                }
+                // TODO: can we detect that we fail phase 2 from missing messages?
+        } else if (3 == msg_params->test_phase) { // Tests to see if even with a ring overflow it can still handle the messages
+                if (*((int *)msg_data) != msg_params->test_msg_count) {
+                        printf("FAILED TEST 3: received %d instead of %d\n", *((int *)msg_data), msg_params->test_msg_count);
+                        printf("---------------------------\n");
+                        msg_params->tests_failed++;
+                        msg_params->test_phase++;
+                } else {
+                        msg_params->test_msg_count++;
+                }
+                rte_free(msg_data);
+                if ((int)(rte_ring_get_size(msg_params->msg_q) - 1) == msg_params->test_msg_count) {
+                        if (rte_ring_count(msg_params->msg_q) != 0) {
+                                printf("FAILED TEST 3: Shouldn't be any messages left, but there are %d in the ring. This may cause future tests to fail.\n", rte_ring_count(msg_params->msg_q));
+                                printf("---------------------------\n");
+                                while (rte_ring_count(msg_params->msg_q) != 0) {
+                                        rte_ring_dequeue(msg_params->msg_q, msg_data);
+                                }
+                                msg_params->tests_failed++;
+                                msg_params->test_phase++;
+                        } else {
+                                printf("PASSED TEST 3\n");
+                                printf("---------------------------\n");
+                                msg_params->tests_passed++;
+                                msg_params->test_phase++;
+                                msg_params->test_msg_count = 0;
+                        }
+                }
+        }
+        if (4 == msg_params->test_phase) {
+                printf("TEST 4: Memory Deallocation...\n");
```

Comment:
This test is based on the results of other tests, which can become problematic. Could you maybe perform both the allocation and deallocation in the same test, and verify that functionality altogether?